src/unistd.rs

```diff
@@ -375,16 +377,23 @@ pub fn sleep(seconds: libc::c_uint) -> c_uint {
 }
 
 #[inline]
-pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<RawFd> {
-    let res = try!(template.with_nix_path(|path| {
-        let mut path_copy = path.to_bytes_with_nul().to_owned();
-        let c_template: *mut c_char = path_copy.as_mut_ptr() as *mut c_char;
+pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
```

Comment:
Yeah, I don't much like that. Mutating the input parameter feels like a hack and a bit of a safety hazard (though the `mut` mark makes it a bit less terrifying), to be honest. Plus, `CStr` input parameters are always kind of messy - I just had reason to use `memfd_create`, and its signature is kind of a pain.